Samples/Samples/View/CalendarPage.xaml.cs

```diff
@@ -1,10 +1,25 @@
-﻿namespace Samples.View
+﻿using System;
+using System.Collections.ObjectModel;
+using Xamarin.Essentials;
+using Xamarin.Forms;
+
+namespace Samples.View
 {
     public partial class CalendarPage : BasePage
     {
         public CalendarPage()
         {
             InitializeComponent();
         }
+
+        void OnEventTapped(object sender, ItemTappedEventArgs e)
+        {
+            if (e.Item == null)
+                return;
+
+            var modal = new CalendarEventPage();
+            modal.BindingContext = e.Item as IEvent;
```

Comment:
I would probably move this cast up to the if statement, as we probably don't want to show an empty page in case the `e.Item` is not an `IEvent`.